### PR TITLE
feat[cli]: only export linker information for processed images

### PR DESCRIPTION
**Pull request by peshence**
attempt at solving the second definition of done cirteria, closes https://github.com/UCLH-Foundry/the-rolling-skeleton/issues/250

### CODE REVIEW

File: cli/src/pixl_cli/_database.py [hunk 1]
```diff
@@ -114,11 +114,15 @@ def _add_new_image_to_session(extract: Extract, message: Message, session: Sessi
...
     """Given a project, get all images in the DB for that project."""
     PixlSession = sessionmaker(engine)
     with PixlSession() as session:
         return cast(
             list[Image],
-            session.query(Image).join(Extract).filter(Extract.slug == project_slug).all(),
+            session.query(Image)
+            .join(Extract)
+            .filter(Extract.slug == project_slug)
+            .filter(not_(Image.pseudo_study_uid.is_(None)))
```
Comment (stefpiatek) [inline cli/src/pixl_cli/_database.py:126]: ooh avoiding the `== None` is fun. Might be worth making a separate PR where you find any time we've done that and replace with nicer syntax

File: cli/src/pixl_cli/_database.py [hunk 2]
```diff
@@ -114,11 +114,15 @@ def _add_new_image_to_session(extract: Extract, message: Message, session: Sessi
     return new_image
 
 
-def images_for_project(project_slug: str) -> list[Image]:
+def processed_images_for_project(project_slug: str) -> list[Image]:
```
Comment (stefpiatek) [inline cli/src/pixl_cli/_database.py:117]: would be good to have a test which ensures that this works as we'd expect (and stops us reverting this behaviour)

File: cli/src/pixl_cli/_database.py (hunk 1 above)
Comment (stefpiatek) [reply cli/src/pixl_cli/_database.py:126]: Also think that maybe this should be where `exported_at` isn't empty instead of `pseudo_study_uid`. Feels like you want to know when something has been exported and that you should expect to find it

**Review by stefpiatek — APPROVED** [review]
Looks good, thanks for finishing this off 🎉 

Some small suggestions

File: cli/src/pixl_cli/_database.py (hunk 1 above)
Comment (peshence) [reply cli/src/pixl_cli/_database.py:126]: oof good catch
